File: Demo/OC/AppFrameworks/UIKit/YJUINavigationRouter/YJUINavigationRouter/Main/Main/YJMainViewController.m
//
//  YJMainViewController.m
//  YJUINavigationRouter
//
//  Created by admin on 2016/12/20.
//  Copyright © 2016年 YJCocoa. All rights reserved.
//

#import "YJMainViewController.h"
#import "YJRouteHeader.h"

@interface YJMainViewController () <YJURLRouterProtocol>
@end

@implementation YJMainViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.title = @"YJMainViewController";
    YJLogDebug(@"初始化 YJMainViewController");
}

- (void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event {
    [YJURLRouter.shared openURLWithUrl:YJRouterURLOther options:@{@"1":@"阳君"} completion:nil];
}

#pragma mark - YJNSURLRouterProtocol
- (void)routerReloadDataWith:(NSDictionary<NSString *,id> *)options completion:(void (^)(NSDictionary<NSString *,id> * _Nonnull))handler {
    YJLogDebug(@"YJMainViewController 接收数据：%@", options);
}

@end

File: Demo/Swift/AppFrameworks/UIKit/YJURLRouter/YJURLRouter/YJMainViewController.swift
//
//  YJMainViewController.swift
//  YJURLRouter
//
//  Created by 阳君 on 2019/6/4.
//  Copyright © 2019 YJCocoa. All rights reserved.
//

import UIKit
import YJCocoa

class YJMainViewController: UIViewController {
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.title = "YJMainViewController"
        YJLogDebug("初始化 YJMainViewController")
    }
    
    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
        YJURLRouter.shared.openURL(url: YJRouterURLOther, options: ["1":"阳君"]) { (options: Dictionary<String, Any>) in
            YJLogDebug("接收回调数据：\(options)")
        }
    }
    
    override func routerReloadData(with options: Dictionary<String, Any>, completion handler: YJRCompletionHandler?) {
        YJLogDebug("YJMainViewController 接收数据：\(options)")
    }
    
}
